src/algorithms/standard/vectorrealtotensor.h

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2006-2016  Music Technology Group - Universitat Pompeu Fabra
+ *
+ * This file is part of Essentia
+ *
+ * Essentia is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU Affero General Public License as published by the Free
+ * Software Foundation (FSF), either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the Affero GNU General Public License
+ * version 3 along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+
+#ifndef ESSENTIA_VECTORREALTOTENSOR_H
+#define ESSENTIA_VECTORREALTOTENSOR_H
+
+#include "streamingalgorithm.h"
+#include "vectoroutput.h"
+
+namespace essentia {
+namespace streaming {
+
+class VectorRealToTensor : public Algorithm {
+ protected:
+  Sink<std::vector<Real> > _frame;
+  Source<Tensor<Real> > _tensor;
+
+  std::vector<int> _shape;
+  size_t _timeStamps;
+  size_t _batchHopSize;
+  size_t _patchHopSize;
+  bool _push;
+  bool _accumulate;
+  bool _pushedEverything;
+  std::string _lastPatchMode;
+
+  std::vector<std::vector<std::vector<Real> > > _acc;
+
+ public:
+  VectorRealToTensor(){
+    declareInput(_frame, 128,"frame", "the input frames");
+    declareOutput(_tensor, 1, "tensor", "the accumulated frame in one single tensor");
+
+  }
+
+  void declareParameters() {
+    // TODO: set a better default shape
+    std::vector<int> outputShape = {-1, 1, 128, 128};
+    declareParameter("shape", "the size of output tensor. If batch dimension (the first one) is -1 it will accumulate as many batches as available in the input track", "", outputShape);
```

Comment:
We can add more hints on what dimensions mean: "the size of output tensor (batch size, number of channels, ... )"

Reply:
"If batch dimension (the first one) is -1 it will accumulate as many batches as available in the input track" -- rephrase

Reply:
"the size of the output tensor (batchsize, arg2, arg3, arg4). If batchsize is -1, all batches from the input track will be (used/gathered)" ?